src/nsls2api/api/v1/beamline_api.py

```diff
@@ -83,16 +84,47 @@ async def add_detector(name: str, detector: Detector):
         beamline_name=name,
         detector_name=detector.name,
         directory_name=detector.directory_name,
+        granularity=detector.granularity,
+        description=detector.description,
+        manufacturer=detector.manufacturer,
     )
 
     if new_detector is None:
         raise HTTPException(
             status_code=409,
             detail=f"Detector {detector.name} already exists in beamline {name}",
-    )
+        )
 
     return new_detector
 
+
+@router.put(
```

Comment:
What do we make then of the fact that RFC more-or-less forbids submitting a request body with a DELETE request? In fact, httpx will not let you do so unless you manually build the request.

Reply:
What are we including in the request body ?

Reply:
if detector name was guaranteed to be unique, could you just do a DELETE request with the detector name alone? as it is now, only the directory_name is guaranteed to be unique.

https://github.com/nmaytan/nsls2api/blob/c1932bb0b0fa0f914fa9e7cfde0b60fbadffd95a/src/nsls2api/services/beamline_service.py#L123-L130

Reply:
The request body right now contains the detector you wish to delete, mirroring the endpoint which adds the detector.

I'm in agreement that perhaps the thing to do is have a endpoint per-detector, and use DELETE/PUT semantics on the URI instead.

We can enforce a unique detector name, we likely want that anyway.

Reply:
I've refactored these two endpoints into one which behaves in accordance with the method used. 

I'm unsure if there's a more graceful way to handle an optional request body than re-raising the `422` when it should be required. It would be nice if it was, since this is slightly misleading (makes it seem optional to the `PUT` and makes it seem acceptable to the `DELETE`).